dataland-frontend/src/components/forms/parts/elements/basic/UploadDocumentsForm.vue

```diff
@@ -102,21 +103,30 @@ export default defineComponent({
      * @param event full event object containing the files
      * @param event.files files
      */
-    async handleFilesSelected(event: FileUploadSelectEvent): Promise<void> {
+    handleFilesSelected(event: FileUploadSelectEvent) {
+      console.log("wyvhodzi 0", event.files);
       const selectedFilesByUser = event.files as File[];
       if (isThereActuallyANewFileSelected(selectedFilesByUser, this.documentsToUpload)) {
-        const lastSelectedFile = selectedFilesByUser[selectedFilesByUser.length - 1];
-        const documentToUpload = { file: lastSelectedFile } as DocumentToUpload;
-        documentToUpload.reference = await calculateSha256HashFromFile(documentToUpload.file);
-        documentToUpload.fileNameWithoutSuffix = removeFileTypeExtension(documentToUpload.file.name);
-        this.documentsToUpload.push(documentToUpload);
-        this.emitDocumentsChangedEvent();
+        const documentsToUpload = Promise.all(
+          selectedFilesByUser.map(async (file) => {
+            return {
+              file: file,
+              reference: await calculateSha256HashFromFile(file),
+              fileNameWithoutSuffix: removeFileTypeExtension(file.name),
+            };
+          }),
```

Comment:
There was a reason why this `UploadDocumentsForm` was originally implemented in a way, that it allows the user to only select one file at a time.   It avoided a lot of cornercases and weird bugs.

Now you have adjusted it to be able to select multiple files at a time.

Have you made sure that there are no bugs?  The tests should detect most of them, but not all.

Have you played around with it a little bit and tested specific cases, e.g.
- multiple files with same content but different names
- multiple files with same content and names
- multiple files with different content and same names etc.

Reply:
Checking ...

Reply:
Testest:
* same content different names - Uploaded
* different content different names - Uploaded
* different content same names - unable to select at the same time. Impossible to have files with same name in same folder. Tested with multiple clicks - Not allowed... popup with warning
* same content same names -  - unable to select at the same time. Impossible to have files with same name in same folder. Tested with multiple clicks - Not allowed... popup with warning

Looks good 👍

Reply:
All right. Thanks for making sure.   Just had the memory that it was rather error-prone back then when we first introduced this.
Closing this since you made sure it works.